src/main/java/cloud/filibuster/instrumentation/libraries/lettuce/RedisInterceptor.java

```
@@ -0,0 +1,152 @@
+
+        String redisMethodName = invocation.getMethod().getName();
+        logger.log(Level.INFO, "methodName: " + redisMethodName);
+
+        // ******************************************************************************************
+        // Construct preliminary call site information.
+        // ******************************************************************************************
+
+        CallsiteArguments callsiteArguments = new CallsiteArguments(invocation.getClass(), Arrays.toString(invocation.getArguments()));
+
+        Callsite callsite = new Callsite(redisConnectionString, REDIS_MODULE_NAME, redisMethodName, callsiteArguments);
+
+        FilibusterClientInstrumentor filibusterClientInstrumentor = new FilibusterClientInstrumentor(redisConnectionString, shouldCommunicateWithServer(), contextStorage, callsite);
+
+        filibusterClientInstrumentor.prepareForInvocation();
+
+        // ******************************************************************************************
+        // Record invocation.
+        // ******************************************************************************************
+
+        filibusterClientInstrumentor.beforeInvocation();
+
+        // ******************************************************************************************
+        // Attach metadata to outgoing request.
+        // ******************************************************************************************
+
+        logger.log(Level.INFO, "requestId: " + filibusterClientInstrumentor.getOutgoingRequestId());
+
+        // ******************************************************************************************
+        // Get forcedException information.
+        // ******************************************************************************************
+
+        JSONObject forcedException = filibusterClientInstrumentor.getForcedException();
```

I know we're not using failure metadata -- probably useful to actually check and throw if it's set to indicate to the user that the option is not supported instead of silently fail.  It's always better to throw instead of silently failing.